tools/src/test/java/org/terracotta/utilities/io/FilesTestBase.java

```diff
@@ -711,6 +714,10 @@ protected enum FileType {
     private final Path linkTarget;
 
     public PathDetails(Path root, Path path, boolean followLinks) throws IOException {
+      FileStore fileStore = getFileStore((java.nio.file.Files.exists(path) ? path : root));
```

Comment:
Shouldn't this be a traversal back up the path to the first "existing" path?

Reply:
If this were a general purpose utility, sure ... but, in this context, it's not necessary -- root is what is needed.